src/gui/torrentcontenttreeview.cpp

```diff
@@ -107,6 +108,78 @@ void TorrentContentTreeView::keyPressEvent(QKeyEvent *event)
     }
 }
 
+void TorrentContentTreeView::setupDownloadPriorityMenu(QMenu *menu, const bool createSubMenu) const
+{
+    const auto model = qobject_cast<TorrentContentFilterModel *>(this->model());
+    Q_ASSERT(model);
+
+    const auto getSelectedRows = [this]() { return selectionModel()->selectedRows(TorrentContentModelItem::COL_PRIO); };
+
+    const auto applyPriorities = [&](const BitTorrent::DownloadPriority priority)
+    {
+        return [model, getSelectedRows, priority]()
```

Comment:
I would sort the capture list alphabetically.

Reply:
Why all this "coolness" here? Why have a lambda that returns another lambda? Is this required here?

Reply:
> I would sort the capture list alphabetically.

Fixed by 251eb8331eb1b00057ded6d8089e257de9b09ff7

Reply:
>Is there something fundamentally wrong with lambdas returning lambdas?

Fundamentally, no. But in this particular case, it doesn't seem to make sense - the inner lambda can be easily omitted, and all its logic moved to the outer one, isn't that right?